clj/test/cljd/test_clojure/test_test.cljd

```diff
@@ -0,0 +1,42 @@
+(ns cljd.test-clojure.test-test
+  (:use [cljd.test :only [deftest testing is use-fixtures]]))
+
+(def ^:private test-state
+  (atom {}))
+
+(defn ^:private onetime-fixture
+  ([] ;; setup
+   (swap! test-state assoc :base-data :foo))
+  ;; the "teardown" (1-arity) version is called with what the "setup" returned
+  ([_] ;; teardown
+   (reset! test-state {})))
+
+(defn ^:private repeated-fixture
+  "Resets test-state to what it was before running the tests"
+  ([] ;;setup
+   @test-state)
+  ([start-state] ;; teardown
+   (reset! test-state start-state)))
+
+(use-fixtures :once onetime-fixture)
+(use-fixtures :each repeated-fixture)
+
+(deftest mutator-test
+  (testing "Initial state"
+    (is (= {:base-data :foo} @test-state)
+        ":once fixtures are run before any tests and in the very end"))
+
+  (testing "Mutating state"
+    (swap! test-state assoc :base-data :bar)
+    (is (= :bar (:base-data @test-state)))
+
+    (swap! test-state assoc :my-data :baz)
+    (is (= :baz (:my-data @test-state))))
+
+  (testing "Mutated state"
+    (is (= :baz (:my-data @test-state))
+        ":each fixtures are run around `deftest`, not individual `is` or `testing`")))
+
+(deftest control-test
```

Comment:
One issue here is that we have no guarantee on test execution order. Thus control-test may run first.

We can work around this by having two "mutator-test", each ensuring the test is in its initial state at the start. By doing so we will be certain that one of the two is run after the other (it sounds stupid put like that but 🤷‍♂️).

Reply:
>we have no guarantee on test execution order

I had no idea. Adjusted the tests as you suggested. I've also run the tests swapping them around manually to check they're independent of order.